File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math


def solution(progresses, speeds):
    answer = []
    checked = []

    for i in range(len(progresses)):
        needPoint = 100 - progresses[i]
        needDay = math.ceil(needPoint / speeds[i])
        checked.append(needDay)

    checked.reverse()

    while len(checked) != 0:
        topPoint = checked[-1]
        count = 1
        checked.pop()
        while len(checked) != 0 and topPoint >= checked[-1]:
            count += 1
            checked.pop()

        answer.append(count)

    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];

    const n = progresses.length
    const days = progresses.map((progress, index) => Math.ceil((100 - progress) / speeds[index]))

    let count = 0;
    let maxDays = days[0]

    for (let i = 0; i < n; i++) {
        if (days[i] <= maxDays) {
            count++
        } else {
            answer.push(count)
            count = 1
            maxDays = days[i]
        }
    }
    answer.push(count)
    return answer;
}